jest.config.ts

```diff
@@ -37,16 +37,84 @@ function jestGithubRunnerSpecs(): JestConfig {
   };
 }
 
+interface ShardConfig {
+  patterns: string[];
+  threshold?: {
+    branches?: number;
+    functions?: number;
+    lines?: number;
+    statements?: number;
+  };
+}
+
+type JestShardedSubconfig = Pick<
```

Comment:
### What about our own issue?

Does your `TEST_SHARD` thing help (or close) this issue?

- https://github.com/renovatebot/renovate/issues/14572

Reply:
I didn't even try because the most recent comment in this issue starts with 'This is a big problem for us'. Instead, we can achieve the same effect by specifying shards manually and generating a couple of Jest config fields from them.

Reply:
> I didn't even try because the most recent comment in this issue starts with 'This is a big problem for us'.

Yeah, trying `jest --shard` is probably a bad idea, seeing as the coverage collection bug has not been fixed (yet). So I understand you skipping over that issue thread. 😄 

> Instead, we can achieve the same effect by specifying shards manually and generating a couple of Jest config fields from them.

This sounds really cool and clever! Do you think we should close #14572 with your PR then?

Reply:
> This sounds really cool and clever! 

Thank you 😊

> Do you think we should close https://github.com/renovatebot/renovate/issues/14572 with your PR then?

Yes, the PR title already updated